### PR TITLE
media-libs/mesa: Add "X" USE flag to enable/disable X11

### PR DESCRIPTION
**Pull request by zamaudio**
This patch came mostly from dolphinling's patch on bugzilla #608912
but I modified it to bring it up to mesa version 17.0.3

Please check this patch carefully as I might have made a mistake.

Signed-off-by: Damien Zammit <damien@zamaudio.com>

### CODE REVIEW

**Comment by zamaudio** [comment]
See second attempt above in force push ^^ thanks.

File: media-libs/mesa/mesa-17.0.3.ebuild [hunk 1]
```diff
@@ -42,21 +42,23 @@ done
...
 	llvm?   ( gallium )
 	opencl? ( gallium llvm )
-	openmax? ( gallium )
+	openmax? ( X gallium )
 	gles1?  ( egl )
 	gles2?  ( egl )
-	vaapi? ( gallium )
-	vdpau? ( gallium )
+	vaapi? ( X gallium )
+	vdpau? ( X gallium )
```
Comment (mattst88) [inline media-libs/mesa/mesa-17.0.3.ebuild]: I'm not sure why openmax/vaapi/vdpau would require X. Do they not compile without some X dependency?

**Comment by mattst88** [comment]
Thanks. These changes looks like what I expect.

I have concerns about reverse dependencies of Mesa relying on X support. I recently added an X use flag to libepoxy, and it broke gtk+ and xorg-server. Mesa has many more reverse dependencies than libepoxy. Doing the legwork to make sure reverse dependencies aren't broken is unfortunately the majority of the work that has to be done to solve this.

File: media-libs/mesa/mesa-17.0.3.ebuild (hunk 1 above)
Comment (victoredwardocallaghan) [inline media-libs/mesa/mesa-17.0.3.ebuild]: yep indeed, no need for these as I checked.

**Comment by DmitryHetman** [comment]
Please add the same flag to mesa-9999.ebuild

**Comment by DmitryHetman** [comment]
Fix +X to X, it's must be disabled by default, use +wayland instead.

**Comment by zamaudio** [comment]
@mattst88 Can you please advise what exactly I need to do to move forward on this issue? Thanks

**Comment by Moiman** [comment]
@zamaudio You need to check if packages depending on mesa need this flag enabled.
Here is list of packages that depend on mesa 
runtime dependency: https://qa-reports.gentoo.org/output/genrdeps/rindex/media-libs/mesa
build-time dependency: https://qa-reports.gentoo.org/output/genrdeps/dindex/media-libs/mesa

**Comment by candrews** [comment]
_bump_ What's the latest news on this one?

**Comment by zamaudio** [comment]
I don't have time to check all the reverse deps, so I guess this patch is going to sit in the shed.

**Comment by mgorny** [comment]
Closing due to no progress.

**Pull request closed without merging**